platform/backend/src/database/schemas/internal-mcp-catalog.ts

```diff
@@ -27,6 +29,51 @@ const internalMcpCatalogTable = pgTable("internal_mcp_catalog", {
       }>
     >()
     .default([]),
+  // Server type and remote configuration
+  serverType: text("server_type").$type<"local" | "remote">(),
+  serverUrl: text("server_url"), // For remote servers
+  docsUrl: text("docs_url"), // Documentation URL for remote servers
+  userConfig: jsonb("user_config")
+    .$type<
+      Record<
+        string,
+        {
+          type: "string" | "number" | "boolean" | "directory" | "file";
+          title: string;
+          description: string;
+          required?: boolean;
+          default?: string | number | boolean | Array<string>;
+          multiple?: boolean;
+          sensitive?: boolean;
+          min?: number;
+          max?: number;
+        }
+      >
+    >()
+    .default({}),
+  // OAuth configuration for remote servers
+  oauthConfig: jsonb("oauth_config").$type<{
+    name: string;
+    server_url: string;
+    auth_server_url?: string;
+    resource_metadata_url?: string;
+    client_id: string;
+    client_secret?: string;
+    redirect_uris: Array<string>;
+    scopes: Array<string>;
+    description?: string;
+    well_known_url?: string;
+    default_scopes: Array<string>;
+    supports_resource_metadata: boolean;
+    generic_oauth?: boolean;
+    token_endpoint?: string;
+    access_token_env_var?: string;
+    requires_proxy?: boolean;
+    provider_name?: string;
+    browser_auth?: boolean;
+    streamable_http_url?: string;
+    streamable_http_port?: number;
+  }>(),
```

Comment:
should we have a separate table for the secrets, so it will be easier to implement other storage types in the future? See example https://docs.keephq.dev/deployment/secret-store#db-secret-manager